File: MercaderSG/Extension.vb
Module Extension
    <System.Runtime.CompilerServices.Extension()> _
    Public Function Paginacion(Of TSource)(source As IEnumerable(Of TSource), Pagina As Integer) As IEnumerable(Of TSource)
        Const CantidadRegPag As Integer = 25
        Return source.Skip((Pagina - 1) * CantidadRegPag).Take(CantidadRegPag)
    End Function
End Module

File: MercaderSG/Extension.cs
using System.Collections.Generic;
using System.Linq;

namespace MercaderSG
{
    static class Extension
    {
        public static IEnumerable<TSource> Paginacion<TSource>(this IEnumerable<TSource> source, int Pagina)
        {
            const int CantidadRegPag = 25;
            return source.Skip((Pagina - 1) * CantidadRegPag).Take(CantidadRegPag);
        }
    }
}